packages/composite-checkout/src/components/wpcom/checkout.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { Checkout, CheckoutProvider } from '../../public-api';
+import { OrderReview } from './order-review';
+
+// These are used only for non-redirect payment methods
+const onSuccess = () => console.log( 'Payment succeeded!' );
+const onFailure = error => console.error( 'There was a problem with your payment', error );
+
+// These are used only for redirect payment methods
+const successRedirectUrl = window.location.href;
+const failureRedirectUrl = window.location.href;
+
+export function callWpcomShoppingCartEndpoint() {
+	// return await wpcom.req.get( '/me/stripe-configuration', query );
+	// something like this
+}
+
+// This is the parent component which would be included on a host page
+export function WPCOMCheckout( { useShoppingCart, availablePaymentMethods } ) {
+	const { itemsWithTax, total, deleteItem, changePlanLength } = useShoppingCart();
+
+	const ReviewContent = () => (
+		<OrderReview
+			items={ itemsWithTax }
+			total={ total }
+			onDeleteItem={ deleteItem }
+			onChangePlanLength={ changePlanLength }
+		/>
+	);
+
+	const handleCheckoutEvent = ( { type, payload }, dispatch, next ) => {
+		console.log( 'WPCOM checkout event captured: ' + event );
+	};
+
+	return (
+		<CheckoutProvider
+			locale={ 'US' }
+			items={ itemsWithTax }
+			total={ total }
+			onSuccess={ onSuccess }
+			onFailure={ onFailure }
+			successRedirectUrl={ successRedirectUrl }
+			failureRedirectUrl={ failureRedirectUrl }
+			eventHandler={ handleCheckoutEvent }
```

Comment:
You can remove this, btw. The new way to do this is to use `createRegistry()`, pass the result to `CheckoutProvider` as a the `registry` prop, and then use `registry.subscribe()` to subscribe to all changes to the store(s) (see the main demo for an example). That can probably wait, though, until we need to do something with those events.